Improve documentation and error messaging for .at method with multiple indices (#1595)
PennyLaneAI/catalyst: doc/dev/jax_integration.rst

  more details, see `jax.numpy.ndarray.at
  <https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.ndarray.at.html>`__.

  .. note::

      When using multiple indices with the `.at` method, the indices must be sorted and unique.

>       However, these requirements cannot be checked at compile time, and one must explicitly provide

Reviewer: I think we tend to prefer "you" when giving instructions in the docs, rather than "one". cc @josh146

```suggestion
      However, these requirements cannot be checked at compile time, therefore you must explicitly provide
```

- ``jax.numpy.polyfit``
- ``jax.numpy.fft``

> - ``jax.numpy.ndarray.at[index]`` when ``index`` corresponds to multiple
>   indices. Note that updating multiple indices is supported as long as the
>   indices are sorted and unique, and the appropriate parameters are provided
>   (see the :ref:`jax-sharp-bits` section for details).

Reviewer 2: I think the previous entry is important to preserve, unless we support supplying all indices now? (can't remember)

For the new entry:
```suggestion
- ``jax.numpy.ndarray.at[index]``  when ``index`` corresponds to all array
  indices.
- ``jax.numpy.ndarray.at[index]``  when ``index`` contains multiple indices
  *and* the indices overlap, or are not sorted in ascending order (see the
  :ref:`jax-sharp-bits` section for details).
```

Reviewer 2: Neat 👍

- ``jax.numpy.polyfit``
- ``jax.numpy.fft``
- ``jax.numpy.ndarray.at[index]``  when ``index`` contains multiple indices
  *and* the indices overlap, or are not sorted in ascending order (see the

>   :ref:`jax-sharp-bits` section for details).

Author: yes, I am aware of that, but not sure how to resolve it 😅. what I did looks fine to me, do you have any idea what is going wrong?

>   :ref:`jax-sharp-bits` section for details).

Reviewer: Looks like it's resolved now with the addition of `.. _jax-sharp-bits:` :slightly_smiling_face:
